Add ADP-specific glossary terms for GA (#1691)
redpanda-data/documentation: modules/terms/partials/oauth-client.adoc

=== OAuth Client

> :term-name: OAuth Client

Author: ```suggestion
=== OAuth client
:term-name: OAuth client
```

> :hover-text: An admin-registered external tool (such as Claude.ai, ChatGPT, or Cursor) authorized to request tokens from the in-dataplane OAuth Authorization Server. Distinct from an OAuth Provider.

Author: ```suggestion
:hover-text: An admin-registered external tool (such as Claude.ai, ChatGPT, or Cursor) authorized to request tokens from the in-dataplane OAuth Authorization Server. Distinct from an OAuth provider.
```

> :hover-text: An admin-registered external tool (such as Claude.ai, ChatGPT, or Cursor) authorized to request tokens from the in-dataplane OAuth Authorization Server. Distinct from an OAuth provider.

Author: ```suggestion
:hover-text: An admin-registered external tool (such as Claude.ai, ChatGPT, or Cursor) authorized to request tokens from the in-dataplane OAuth authorization server. Distinct from an OAuth provider.
```

> :hover-text: An ADP resource that governs inbound authentication: an external app (Claude Desktop, ChatGPT, Copilot Studio) authenticating to AI Gateway so its users can invoke MCP tools. Separate from an OAuth provider, which governs outbound authentication.

Reviewer: ```suggestion
:hover-text: An ADP resource that governs inbound authentication: an external app (Claude Desktop, ChatGPT, Copilot Studio) authenticating to AI Gateway so the app's users can invoke MCP tools. Separate from an OAuth provider, which governs outbound authentication.
```
